sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/models/FileProperties.java

```diff
@@ -4,32 +4,69 @@
 package com.azure.storage.file.models;
 
 import com.azure.core.implementation.util.ImplUtils;
+import com.azure.storage.file.FileSmbProperties;
+
 import java.time.OffsetDateTime;
 import java.util.Map;
 
 /**
  * Contains property information about a File in the storage File service.
  */
 public final class FileProperties {
-    private final String eTag;
-    private final OffsetDateTime lastModified;
-    private final Map<String, String> metadata;
-    private final String fileType;
-    private final Long contentLength;
-    private final String contentType;
-    private final byte[] contentMD5;
-    private final String contentEncoding;
-    private final String cacheControl;
-    private final String contentDisposition;
-    private final OffsetDateTime copyCompletionTime;
-    private final String copyStatusDescription;
-    private final String copyId;
-    private final String copyProgress;
-    private final String copySource;
-    private final CopyStatusType copyStatus;
-    private final Boolean isServerEncrypted;
+    private String eTag;
+    private OffsetDateTime lastModified;
+    private Map<String, String> metadata;
+    private String fileType;
+    private Long contentLength;
+    private String contentType;
+    private byte[] contentMD5;
+    private String contentEncoding;
+    private String cacheControl;
+    private String contentDisposition;
+    private String contentLanguage;
+    private OffsetDateTime copyCompletionTime;
+    private String copyStatusDescription;
+    private String copyId;
+    private String copyProgress;
+    private String copySource;
+    private CopyStatusType copyStatus;
+    private Boolean isServerEncrypted;
+    private FileSmbProperties smbProperties;
+    private String filePermission;
```

Comment:
Should I remove this and make it an extra parameter in the setfileproperties and createfile calls?

Reply:
Doing this to be consistent

Reply:
As discussed, I think it's better to pull it out so it's consistent with directories